src/Extension/FluentExtension.php

```diff
@@ -654,6 +666,16 @@ public function onAfterDuplicate($original, $doWrite, $relations): void
                     SELECT ? AS \"RecordID\", \"Locale\", $fields_str
                     FROM \"$localisedTable\"
                     WHERE \"RecordID\" = ?", [$toID, $fromID]);
+
+            // Make sure to update any has_one IDs - otherwise the new localised table entry will
+            // have the has_one ID from the original record.
+            $fieldsToUpdate = [];
+            foreach (array_intersect($fields, $copyHasOneRelations) as $copyFieldName) {
+                $fieldsToUpdate[$copyFieldName] = $owner->$copyFieldName;
+            }
+            if (!empty($fieldsToUpdate)) {
+                SQLUpdate::create($localisedTable, $fieldsToUpdate, ['RecordID' => $toID])->execute();
```

Comment:
This is technically functional but I recommend something that is more ORM friendly - an approach similar to what we already have in `onBeforeWrite`. In our case we should be able to use `onBeforeDuplicate` extension point to achieve this.

Reply:
I'm using a low-level API intentionally to mirror the `insert` query above. Ideally setting these values would all be done in a single call, but I wasn't sure how to merge using the correct values into the existing `insert this row based on that existing row` query.

I'm not sure how `onBeforeDuplicate` applies here - can you please explain further?

Reply:
Tried out a different approach using ORM as raw queries should be used only when there is no other option. It turns out that ORM approach has some other issues so I think your approach is overall better. Happy to keep this as is.